### DISCUSSION TITLE
Cannot read property 'validate' of undefined when 'map' MST 'create'

### DISCUSSION THREAD

**Discussion by davetapley** [mobxjs/mobx#2797] [Q&A] [answered]
This question is about `mobx-state-tree`, I hope it's okay here 🙏🏻 
_https://github.com/mobxjs/mobx-state-tree doesn't have GitHub Discussions enabled_

I'm new to this library and integrating with an existing React app.

I have my first model and I can `create` from existing objects:

```
export const StaticLayer = types.model({...

const one = StaticLayer.create(config.layers.static[0])
const two = StaticLayer.create(config.layers.static[1])
const layers = LayersStore.create({ static: [one, two] });
```

⬆️ that works fine, but when I try to generalize to a `map` ⬇️  I get `Cannot read property 'validate' of undefined` from somewhere deep in `mbox-state-tree`.

```
const staticLayers = config.layers.static.map(StaticLayer.create);
const layers = LayersStore.create({ static: staticLayers });
```

I'm a bit too new to the library to parse it myself so any help would be appreciated 🙏🏻 

Full stack from browser:

<img width="1026" alt="Screen Shot 2021-02-11 at 8 46 04 AM" src="https://user-images.githubusercontent.com/48232/107660817-e5dde580-6c45-11eb-9b2e-a0260557ec5a.png">

---

I also tried this approach ⬇️ and I now get `Error: [mobx-state-tree] Cannot modify 'AnonymousModel[]@/static', the object is protected and can only be modified by using an action.` Maybe a clue?

```
  const layers = LayersStore.create();
  config.layers.static.forEach(l => {
    layers.static.push(l)
  })
```

**Comment by mweststrate** [comment, marked as answer]
Probably you'll need `config.layers.static.create(data => StaticLayer.create(data))` instead. For your last errors, make sure to familiarize yourself with actions as described in the documentation. For further questions MST has it's own community living over here: https://spectrum.chat/mobx-state-tree
